### PR TITLE
wasm: add wasmEdge container process to cgroup

### PR DESCRIPTION
**Pull request by Poorunga**
Add wasmEdge container process to cgroup, follow up #74

### CODE REVIEW

File: wasm/src/wasmedge.rs [hunk 1]
```diff
@@ -263,9 +276,7 @@ impl ProcessLifecycle<ExecProcess> for WasmEdgeExecLifecycle {
     }
 
     async fn delete(&self, _p: &mut ExecProcess) -> containerd_shim::Result<()> {
-        Err(Error::Unimplemented(
```
Comment (Burning1020) [inline wasm/src/wasmedge.rs:266]: Exec in wasm container is still not supported, so should return `Err` here.

File: wasm/src/utils.rs [hunk 2]
```diff
@@ -75,3 +76,23 @@ pub fn get_memory_limit(spec: &Spec) -> Option<i64> {
         .and_then(|x| x.memory().as_ref())
         .and_then(|x| x.limit())
 }
+
+pub(crate) fn get_rootfs(spec: &Spec) -> containerd_shim::Result<String> {
```
Comment (Burning1020) [inline wasm/src/utils.rs:80]: Better to return `Option`, let the caller decide how to handle `None`

**Review by Burning1020 — CHANGES_REQUESTED** [review]

File: wasm/src/wasmtime.rs [hunk 3]
```diff
@@ -59,16 +59,9 @@ impl ContainerFactory<WasmtimeContainer> for WasmtimeContainerFactory {
...
         spec.canonicalize_rootfs(req.bundle())
             .map_err(|e| Error::InvalidArgument(format!("could not canonicalize rootfs: {e}")))?;
-        let rootfs = spec
-            .root()
-            .as_ref()
-            .ok_or(Error::InvalidArgument(
-                "rootfs is not set in runtime spec".to_string(),
-            ))?
-            .path();
-        mkdir(rootfs, 0o711).await?;
```
Comment (abel-von) [inline wasm/src/wasmtime.rs:69]: why is this mkdir removed?

Comment (Poorunga) [reply wasm/src/wasmtime.rs:69]: Sorry, will update later

**Review by Burning1020 — APPROVED** [review]
LGTM. Good job, thank you.

File: wasm/src/wasmedge.rs (hunk 1 above)
Comment (Poorunga) [reply wasm/src/wasmedge.rs:266]: done

File: wasm/src/utils.rs (hunk 2 above)
Comment (Poorunga) [reply wasm/src/utils.rs]: done

**Pull request merged**